app/models/principal.rb

```diff
@@ -114,6 +118,10 @@ def self.visible_by_condition(principal)
           project_ids)
   end
 
+  def self.order_by_name
+    order(User::USER_FORMATS_STRUCTURE[Setting.user_format].map(&:to_s))
+  end
```

Comment:
why isn't that a scope? Probably `ordered_by_name`, for nicer reading experience...